specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/networkWatcher.json

```diff
@@ -3380,6 +3400,244 @@
+        "testConfigurations",
+        "sources",
+        "destinations"
+      ],
+      "description": "Describes the connection monitor test."
+    },
+    "ConnectionMonitorTestConfiguration": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the connection monitor test."
+        },
+        "testFrequencySec": {
+          "type": "integer",
+          "description": "The interval between tests in seconds."
+        },
+        "protocol": {
+          "type": "string",
+          "description": "Network protocol.",
+          "enum": [
+            "Tcp",
+            "Http",
+            "Https",
+            "Icmp"
+          ],
+          "x-ms-enum": {
+            "name": "ConnectionMonitorTestConfigurationProtocol",
+            "modelAsString": true
+          }
+        },
+        "httpConfiguration": {
+          "$ref": "#/definitions/ConnectionMonitorHTTPConfiguration",
+          "description": "Describes HTTP configuration of the connection monitor test."
+        },
+        "tcpConfiguration": {
+          "$ref": "#/definitions/ConnectionMonitorTCPConfiguration",
+          "description": "Describes TCP configuration of the connection monitor test."
+        },
+        "icmpConfiguration": {
+          "$ref": "#/definitions/ConnectionMonitorICMPConfiguration",
+          "description": "Describes ICMP configuration of the connection monitor test."
+        },
+        "successThreshold": {
+          "$ref": "#/definitions/ConnectionMonitorSuccessThreshold",
+          "description": "Describes the threshold at which the connection monitor test configuration is considered successful."
+        }
+      },
+      "required": [
+        "name",
+        "protocol"
+      ],
+      "description": "Describes a connection monitor test configuration."
+    },
+    "ConnectionMonitorTCPConfiguration": {
+      "properties": {
+        "disableTraceRoute": {
+          "type": "boolean",
+          "description": "A value indicating whether traceroute is disabled."
+        },
+        "port": {
+          "type": "integer",
+          "description": "The destination port on which a TCP test is to be performed."
+        }
+      },
+      "required": [
+        "port"
+      ],
+      "description": "Describes a configuration for a TCP test."
+    },
+    "ConnectionMonitorICMPConfiguration": {
+      "properties": {
+        "disableTraceRoute": {
+          "type": "boolean",
+          "description": "A value indicating whether traceroute is disabled."
+        }
+      },
+      "description": "Describes a configuration for an ICMP test."
+    },
+    "ConnectionMonitorHTTPConfiguration": {
+      "properties": {
+        "method": {
+          "type": "string",
+          "description": "HTTP method.",
+          "enum": [
+            "Get",
+            "Post"
+          ],
+          "x-ms-enum": {
+            "name": "HTTPConfigurationMethod",
```

Comment:
In generated SDK all Network's enums and models are stored in one directory, so please consider adding prefix `ConnectionMonitor` to this enum